### PR TITLE
Same retry logic pull request, now with passing tests.

### PR DESCRIPTION
**Pull request by paul-krohn**

### CODE REVIEW

File: leadbutt.py [hunk 1]
```diff
@@ -6,8 +6,10 @@
 Options:
   -h --help                   Show this screen.
   -c FILE --config-file=FILE  Path to a YAML configuration file [default: config.yaml].
-  -p INT --period INT         Period length, in minutes (default: 1)
-  -n INT                      Number of data points to try to get (default: 5)
+  -i INTERVAL                 Interval, in ms, to wait between metric requests. Doubles as the backoff multiplier. [default: 50]
+  -m MAX_INTERVAL             The maximum interval time to back off to, in ms [default: 4000]
+  -p INT --period INT         Period length, in minutes [default: 1]
+  -n INT                      Number of data points to try to get [default: 5]
```
Comment (dcrampton) [inline leadbutt.py]: Why the change from parens to brackets?

File: leadbutt.py [hunk 2]
```diff
@@ -144,33 +165,37 @@ def leadbutt(config_file, cli_options, verbose=False, **kwargs):
...
-                metric['Namespace'],  # AWS/ELB, AWS/EC2
-                metric['Statistics'],  # Sum, Maximum
+            results = get_metric_statistics(
+                connection=conn,
+                period=period_local,
+                start_time=start_time,
+                end_time=end_time,
+                metric_name=metric_name,
+                namespace=metric['Namespace'],
+                statistics=metric['Statistics'],
```
Comment (dcrampton) [inline leadbutt.py]: Why the removal of the comments?

File: test_leadbutt.py [hunk 3]
```diff
@@ -161,7 +161,7 @@ def test_can_get_auth_from_config(self, mock_get_config, mock_connect):
                 'aws_secret_access_key': 'bar',
             }
         }
-        leadbutt.leadbutt('dummy_config_file', 'dummy_cli_options')
+        leadbutt.leadbutt('dummy_config_file', {'Count': 1, 'Period': 5})
```
Comment (dcrampton) [inline test_leadbutt.py]: hee hee... leadbutt

**Comment by dcrampton** [comment]
LGTM other than nits, but you will definitely want more eyes on this.

File: leadbutt.py [hunk 4]
```diff
@@ -29,7 +33,7 @@
 else:
     text_type = unicode
 
-__version__ = '0.9.0'
+__version__ = '0.9.2'
```
Comment (tmclaugh) [inline leadbutt.py]: This should be done after the PR is merged.

File: plumbum.py [hunk 5]
```diff
@@ -44,7 +44,7 @@
 import jinja2
 import os.path
 
-__version__ = '0.9.0'
+__version__ = '0.9.2'
```
Comment (tmclaugh) [inline plumbum.py]: This should be done after the PR is merged.

File: leadbutt.py (hunk 1 above)
Comment (paul-krohn) [inline leadbutt.py]: docopt reads the content of the brackets and uses them as defaults.

File: leadbutt.py (hunk 4 above)
Comment (paul-krohn) [inline leadbutt.py]: Agreed, needs to be 0.9.3 anyway.

Comment (tmclaugh) [inline leadbutt.py]: Cool, back those changes out before merging this PR.
